File: 219/js/219.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var containsNearbyDuplicate = function(nums, k) {
    let n = nums.length;
    let check = new Map();

    for (let i = 0; i < n; i++){
        num = nums[i];
        if (check.has(num)) {
            let preId = check.get(num)

            if ((i - preId) <= k) {
                return true;
            }
        }

        check.set(num, i);
    }

    return false;
};

File: 219/py/219.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        n = len(nums)
        check = dict()

        for i in range(0, n):
            num = nums[i]
            
            if num in check:
                preId = check[num]
                if i - preId <= k :
                    return True
            
            check[num] = i

        return False